### PR TITLE
feat: build Field HOC

### PR DESCRIPTION
**Pull request by carloshrabelo**
Task [Link](https://juntossomosmais.monday.com/boards/346135329/pulses/621303379)

This change creates an HOC for Input / Select / TextArea components among other components that will have the same structure within a form.
Enables instantiating the Input / Select components by ref (used in react-hook-form)

### CODE REVIEW

File: packages/react-ds/src/components/Select/Select.stories.mdx [hunk 1]
```diff
@@ -6,7 +6,7 @@ import Spinner from '../Spinner/Spinner'
 import Docs from '@venice/core/docs/Select.md'
 
 import Select from './Select'
-import StoreIcon from '../icons/Store'
+import StoreIcon from '../Icons/Store'
 
 <Meta title="Components|Selects" component={Select} />
```
Comment (felipefialho) [inline packages/react-ds/src/components/Select/Select.stories.mdx:11]: Maybe we can create a new category to Forms?

```
<Meta title="Forms|Selects" component={Select} />
```

File: packages/react-ds/src/hoc/WithField.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+import classNames from 'classnames/bind'
+import uniqid from 'uniqid'
+
+import { IField } from '@venice/core/models/field.model'
+import styles from '@venice/styles/components/Field.module.scss'
+
+import Alert from '../components/Icons/Alert'
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const WithField = (Component: FunctionComponent<any>) =>
```
Comment (felipefialho) [inline packages/react-ds/src/hoc/WithField.tsx:12]: It works?

```ts
interface IWithFieldProps<T> {
   value: T
}

const MyComponent: FunctionComponent<IWithFieldProps<string>> = ({ value }) => <div />;
```

File: packages/react-ds/src/components/Select/Select.stories.mdx (hunk 1 above)
Comment (carloshrabelo) [reply packages/react-ds/src/components/Select/Select.stories.mdx:11]: Then we move Input and Select for there??

File: packages/react-ds/src/hoc/WithField.tsx (hunk 2 above)
Comment (carloshrabelo) [reply packages/react-ds/src/hoc/WithField.tsx:12]: No
"/

File: packages/react-ds/src/components/Input/Input.stories.mdx [hunk 3]
```diff
@@ -7,7 +7,7 @@ import Docs from '@venice/core/docs/Input.md'
 
 import Input from './Input'
 
-<Meta title="Components|Inputs" component={Input} />
+<Meta title="Form|Input" component={Input} />
```
Comment (felipefialho) [inline packages/react-ds/src/components/Input/Input.stories.mdx:10]: `Forms`**

File: packages/react-ds/src/components/Select/Select.stories.mdx [hunk 4]
```diff
@@ -6,9 +6,9 @@ import Spinner from '../Spinner/Spinner'
 import Docs from '@venice/core/docs/Select.md'
 
 import Select from './Select'
-import StoreIcon from '../icons/Store'
+import StoreIcon from '../Icons/Store'
 
-<Meta title="Components|Selects" component={Select} />
+<Meta title="Form|Select" component={Select} />
```
Comment (felipefialho) [inline packages/react-ds/src/components/Select/Select.stories.mdx:11]: `Forms`**

File: packages/react-ds/src/hoc/WithField.tsx (hunk 2 above)
Comment (brunobertolini) [reply packages/react-ds/src/hoc/WithField.tsx:12]: `const WithField = <P extends Object>(Component: FunctionComponent<P>)` not too?

**Comment by brunobertolini** [comment]
@carloshrabelo can you add more description on PR to me (and others) understand the context of this problem/solution

File: packages/react-ds/src/hoc/WithField.tsx (hunk 2 above)
Comment (carloshrabelo) [reply packages/react-ds/src/hoc/WithField.tsx:12]: neither
Screen Shot 2020-06-25 at 13 35 43
Screen Shot 2020-06-25 at 13 37 42
Screen Shot 2020-06-25 at 13 36 28

**Comment by carloshrabelo** [comment]
> @carloshrabelo can you add more description on PR to me (and others) understand the context of this problem/solution?

Updated

**Pull request closed without merging**